--- charts-original/charts/grafana/templates/_helpers.tpl
+++ charts/charts/grafana/templates/_helpers.tpl
@@ -158,6 +158,17 @@
 {{- end -}}
 
 {{/*
+Return the appropriate apiVersion for HorizontalPodAutoscaler.
+*/}}
+{{- define "grafana.hpa.apiVersion" -}}
+  {{- if .Capabilities.APIVersions.Has "autoscaling/v2" }}
+    {{- print "autoscaling/v2" -}}
+  {{- else if .Capabilities.APIVersions.Has "autoscaling/v1" }}
+    {{- print "autoscaling/v1" -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
 Return the appropriate apiVersion for ingress.
 */}}
 {{- define "grafana.ingress.apiVersion" -}}
@@ -167,6 +178,17 @@
     {{- print "networking.k8s.io/v1beta1" -}}
   {{- else -}}
     {{- print "extensions/v1beta1" -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Return the appropriate apiVersion for podDisruptionBudget.
+*/}}
+{{- define "grafana.podDisruptionBudget.apiVersion" -}}
+  {{- if $.Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
+    {{- print "policy/v1" -}}
+  {{- else -}}
+    {{- print "policy/v1beta1" -}}
   {{- end -}}
 {{- end -}}
 
