--- charts-original/charts/grafana/templates/_pod.tpl
+++ charts/charts/grafana/templates/_pod.tpl
@@ -463,10 +463,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.dashboards.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -574,10 +572,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.datasources.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- if .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ .Values.sidecar.skipTlsVerify }}"
@@ -782,10 +778,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.plugins.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.plugins.script }}
       - name: SCRIPT
         value: "{{ . }}"
