--- charts-original/charts/grafana/templates/_pod.tpl
+++ charts/charts/grafana/templates/_pod.tpl
@@ -468,10 +468,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.dashboards.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "kube-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -584,10 +582,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.datasources.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "kube-prometheus-stack.projectNamespaceList" . }}"
       {{- if .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ .Values.sidecar.skipTlsVerify }}"
@@ -792,10 +788,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.plugins.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "kube-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.plugins.script }}
       - name: SCRIPT
         value: "{{ . }}"
