--- charts-original/templates/_pod.tpl
+++ charts/templates/_pod.tpl
@@ -143,10 +143,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.alerts.searchNamespace }}
       - name: NAMESPACE
-        value: {{ . | join "," | quote }}
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.alerts.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: {{ quote . }}
@@ -221,10 +219,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- if .Values.sidecar.datasources.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (.Values.sidecar.datasources.searchNamespace | join ",") . }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -279,10 +275,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.notifiers.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -356,10 +350,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.alerts.searchNamespace }}
       - name: NAMESPACE
-        value: {{ . | join "," | quote }}
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.alerts.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: {{ quote . }}
@@ -468,10 +460,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.dashboards.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -584,10 +574,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.datasources.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- if .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ .Values.sidecar.skipTlsVerify }}"
@@ -688,10 +676,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ .Values.sidecar.enableUniqueFilenames }}"
       {{- end }}
-      {{- with .Values.sidecar.notifiers.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.skipTlsVerify }}
       - name: SKIP_TLS_VERIFY
         value: "{{ . }}"
@@ -792,10 +778,8 @@
       - name: UNIQUE_FILENAMES
         value: "{{ . }}"
       {{- end }}
-      {{- with .Values.sidecar.plugins.searchNamespace }}
       - name: NAMESPACE
-        value: "{{ tpl (. | join ",") $root }}"
-      {{- end }}
+        value: "{{ template "project-prometheus-stack.projectNamespaceList" . }}"
       {{- with .Values.sidecar.plugins.script }}
       - name: SCRIPT
         value: "{{ . }}"
