--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -142,6 +142,11 @@
 
 # Prometheus Operator
 
+{{/* Comma-delimited list of namespaces that need to be watched to configure Project Prometheus Stack components */}}
+{{- define "kube-prometheus-stack.projectNamespaceList" -}}
+{{ append .Values.global.cattle.projectNamespaces .Release.Namespace | uniq | join "," }}
+{{- end }}
+
 {{/* vim: set filetype=mustache: */}}
 {{/* Expand the name of the chart. This is suffixed with -alertmanager, which means subtract 13 from longest 63 available */}}
 {{- define "kube-prometheus-stack.name" -}}
@@ -168,15 +173,10 @@
 {{- end -}}
 {{- end -}}
 
-{{/* Fullname suffixed with -operator */}}
-{{/* Adding 9 to 26 truncation of kube-prometheus-stack.fullname */}}
+{{/* Fullname suffixed with operator */}}
 {{- define "kube-prometheus-stack.operator.fullname" -}}
-{{- if .Values.prometheusOperator.fullnameOverride -}}
-{{- .Values.prometheusOperator.fullnameOverride | trunc 35 | trimSuffix "-" -}}
-{{- else -}}
 {{- printf "%s-operator" (include "kube-prometheus-stack.fullname" .) -}}
 {{- end }}
-{{- end }}
 
 {{/* Prometheus custom resource instance name */}}
 {{- define "kube-prometheus-stack.prometheus.crname" -}}
@@ -245,15 +245,6 @@
 {{- end -}}
 {{- end -}}
 
-{{/* Create the name of kube-prometheus-stack service account to use */}}
-{{- define "kube-prometheus-stack.operator.admissionWebhooks.serviceAccountName" -}}
-{{- if .Values.prometheusOperator.serviceAccount.create -}}
-    {{ default (printf "%s-webhook" (include "kube-prometheus-stack.operator.fullname" .)) .Values.prometheusOperator.admissionWebhooks.deployment.serviceAccount.name }}
-{{- else -}}
-    {{ default "default" .Values.prometheusOperator.admissionWebhooks.deployment.serviceAccount.name }}
-{{- end -}}
-{{- end -}}
-
 {{/* Create the name of prometheus service account to use */}}
 {{- define "kube-prometheus-stack.prometheus.serviceAccountName" -}}
 {{- if .Values.prometheus.serviceAccount.create -}}
@@ -304,17 +295,6 @@
 {{- end -}}
 
 {{/*
-Allow kube-state-metrics job name to be overridden
-*/}}
-{{- define "kube-prometheus-stack-kube-state-metrics.name" -}}
-  {{- if index .Values "kube-state-metrics" "nameOverride" -}}
-    {{- index .Values "kube-state-metrics" "nameOverride" -}}
-  {{- else -}}
-    {{- print "kube-state-metrics" -}}
-  {{- end -}}
-{{- end -}}
-
-{{/*
 Use the kube-state-metrics namespace override for multi-namespace deployments in combined charts
 */}}
 {{- define "kube-prometheus-stack-kube-state-metrics.namespace" -}}
@@ -452,6 +432,25 @@
 {{- end }}
 {{- end -}}
 
+
+{{/* Define ingress for all hardened namespaces */}}
+{{- define "kube-prometheus-stack.hardened.networkPolicy.ingress" -}}
+{{- $root := index . 0 }}
+{{- $ns := index . 1 }}
+{{- if $root.Values.global.networkPolicy.ingress -}}
+{{ toYaml $root.Values.global.networkPolicy.ingress }}
+{{- end }}
+{{- if $root.Values.global.networkPolicy.limitIngressToProject }}
+- from:
+{{- if $root.Values.global.cattle.projectNamespaceSelector }}
+  - namespaceSelector: {{- $root.Values.global.cattle.projectNamespaceSelector | toYaml | nindent 6 }}
+{{- end }}
+  - namespaceSelector:
+      matchLabels:
+        kubernetes.io/metadata.name: {{ $ns }}
+{{- end }}
+{{- end -}}
+
 {{- define "kube-prometheus-stack.operator.admission-webhook.dnsNames" }}
 {{- $fullname := include "kube-prometheus-stack.operator.fullname" . }}
 {{- $namespace := include "kube-prometheus-stack.namespace" . }}
