--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -142,9 +142,14 @@
 
 # Prometheus Operator
 
+{{/* Comma-delimited list of namespaces that need to be watched to configure Project Prometheus Stack components */}}
+{{- define "project-prometheus-stack.projectNamespaceList" -}}
+{{ append .Values.global.cattle.projectNamespaces .Release.Namespace | uniq | join "," }}
+{{- end }}
+
 {{/* vim: set filetype=mustache: */}}
 {{/* Expand the name of the chart. This is suffixed with -alertmanager, which means subtract 13 from longest 63 available */}}
-{{- define "kube-prometheus-stack.name" -}}
+{{- define "project-prometheus-stack.name" -}}
 {{- default .Chart.Name .Values.nameOverride | trunc 50 | trimSuffix "-" -}}
 {{- end }}
 
@@ -155,7 +160,7 @@
 The components in this chart create additional resources that expand the longest created name strings.
 The longest name that gets created adds and extra 37 characters, so truncation should be 63-35=26.
 */}}
-{{- define "kube-prometheus-stack.fullname" -}}
+{{- define "project-prometheus-stack.fullname" -}}
 {{- if .Values.fullnameOverride -}}
 {{- .Values.fullnameOverride | trunc 26 | trimSuffix "-" -}}
 {{- else -}}
@@ -168,62 +173,57 @@
 {{- end -}}
 {{- end -}}
 
-{{/* Fullname suffixed with -operator */}}
-{{/* Adding 9 to 26 truncation of kube-prometheus-stack.fullname */}}
-{{- define "kube-prometheus-stack.operator.fullname" -}}
-{{- if .Values.prometheusOperator.fullnameOverride -}}
-{{- .Values.prometheusOperator.fullnameOverride | trunc 35 | trimSuffix "-" -}}
-{{- else -}}
-{{- printf "%s-operator" (include "kube-prometheus-stack.fullname" .) -}}
-{{- end }}
+{{/* Fullname suffixed with operator */}}
+{{- define "project-prometheus-stack.operator.fullname" -}}
+{{- printf "%s-operator" (include "project-prometheus-stack.fullname" .) -}}
 {{- end }}
 
 {{/* Prometheus custom resource instance name */}}
-{{- define "kube-prometheus-stack.prometheus.crname" -}}
+{{- define "project-prometheus-stack.prometheus.crname" -}}
 {{- if .Values.cleanPrometheusOperatorObjectNames }}
-{{- include "kube-prometheus-stack.fullname" . }}
+{{- include "project-prometheus-stack.fullname" . }}
 {{- else }}
-{{- print (include "kube-prometheus-stack.fullname" .) "-prometheus" }}
+{{- print (include "project-prometheus-stack.fullname" .) "-prometheus" }}
 {{- end }}
 {{- end }}
 
 {{/* Prometheus apiVersion for networkpolicy */}}
-{{- define "kube-prometheus-stack.prometheus.networkPolicy.apiVersion" -}}
+{{- define "project-prometheus-stack.prometheus.networkPolicy.apiVersion" -}}
 {{- print "networking.k8s.io/v1" -}}
 {{- end }}
 
 {{/* Alertmanager custom resource instance name */}}
-{{- define "kube-prometheus-stack.alertmanager.crname" -}}
+{{- define "project-prometheus-stack.alertmanager.crname" -}}
 {{- if .Values.cleanPrometheusOperatorObjectNames }}
-{{- include "kube-prometheus-stack.fullname" . }}
+{{- include "project-prometheus-stack.fullname" . }}
 {{- else }}
-{{- print (include "kube-prometheus-stack.fullname" .) "-alertmanager" -}}
+{{- print (include "project-prometheus-stack.fullname" .) "-alertmanager" -}}
 {{- end }}
 {{- end }}
 
 {{/* Fullname suffixed with thanos-ruler */}}
-{{- define "kube-prometheus-stack.thanosRuler.fullname" -}}
-{{- printf "%s-thanos-ruler" (include "kube-prometheus-stack.fullname" .) -}}
+{{- define "project-prometheus-stack.thanosRuler.fullname" -}}
+{{- printf "%s-thanos-ruler" (include "project-prometheus-stack.fullname" .) -}}
 {{- end }}
 
 {{/* Shortened name suffixed with thanos-ruler */}}
-{{- define "kube-prometheus-stack.thanosRuler.name" -}}
-{{- default (printf "%s-thanos-ruler" (include "kube-prometheus-stack.name" .)) .Values.thanosRuler.name -}}
+{{- define "project-prometheus-stack.thanosRuler.name" -}}
+{{- default (printf "%s-thanos-ruler" (include "project-prometheus-stack.name" .)) .Values.thanosRuler.name -}}
 {{- end }}
 
 
 {{/* Create chart name and version as used by the chart label. */}}
-{{- define "kube-prometheus-stack.chartref" -}}
+{{- define "project-prometheus-stack.chartref" -}}
 {{- replace "+" "_" .Chart.Version | printf "%s-%s" .Chart.Name -}}
 {{- end }}
 
 {{/* Generate basic labels */}}
-{{- define "kube-prometheus-stack.labels" }}
+{{- define "project-prometheus-stack.labels" }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 app.kubernetes.io/version: "{{ replace "+" "_" .Chart.Version }}"
-app.kubernetes.io/part-of: {{ template "kube-prometheus-stack.name" . }}
-chart: {{ template "kube-prometheus-stack.chartref" . }}
+app.kubernetes.io/part-of: {{ template "project-prometheus-stack.name" . }}
+chart: {{ template "project-prometheus-stack.chartref" . }}
 release: {{ $.Release.Name | quote }}
 heritage: {{ $.Release.Service | quote }}
 {{- if .Values.commonLabels}}
@@ -231,46 +231,37 @@
 {{- end }}
 {{- end }}
 
-{{/* Create the name of kube-prometheus-stack service account to use */}}
-{{- define "kube-prometheus-stack.operator.serviceAccountName" -}}
+{{/* Create the name of project-prometheus-stack service account to use */}}
+{{- define "project-prometheus-stack.operator.serviceAccountName" -}}
 {{- if .Values.prometheusOperator.serviceAccount.create -}}
-    {{ default (include "kube-prometheus-stack.operator.fullname" .) .Values.prometheusOperator.serviceAccount.name }}
+    {{ default (include "project-prometheus-stack.operator.fullname" .) .Values.prometheusOperator.serviceAccount.name }}
 {{- else -}}
     {{ default "default" .Values.prometheusOperator.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
 
-{{/* Create the name of kube-prometheus-stack service account to use */}}
-{{- define "kube-prometheus-stack.operator.admissionWebhooks.serviceAccountName" -}}
-{{- if .Values.prometheusOperator.serviceAccount.create -}}
-    {{ default (printf "%s-webhook" (include "kube-prometheus-stack.operator.fullname" .)) .Values.prometheusOperator.admissionWebhooks.deployment.serviceAccount.name }}
-{{- else -}}
-    {{ default "default" .Values.prometheusOperator.admissionWebhooks.deployment.serviceAccount.name }}
-{{- end -}}
-{{- end -}}
-
 {{/* Create the name of prometheus service account to use */}}
-{{- define "kube-prometheus-stack.prometheus.serviceAccountName" -}}
+{{- define "project-prometheus-stack.prometheus.serviceAccountName" -}}
 {{- if .Values.prometheus.serviceAccount.create -}}
-    {{ default (print (include "kube-prometheus-stack.fullname" .) "-prometheus") .Values.prometheus.serviceAccount.name }}
+    {{ default (print (include "project-prometheus-stack.fullname" .) "-prometheus") .Values.prometheus.serviceAccount.name }}
 {{- else -}}
     {{ default "default" .Values.prometheus.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
 
 {{/* Create the name of alertmanager service account to use */}}
-{{- define "kube-prometheus-stack.alertmanager.serviceAccountName" -}}
+{{- define "project-prometheus-stack.alertmanager.serviceAccountName" -}}
 {{- if .Values.alertmanager.serviceAccount.create -}}
-    {{ default (print (include "kube-prometheus-stack.fullname" .) "-alertmanager") .Values.alertmanager.serviceAccount.name }}
+    {{ default (print (include "project-prometheus-stack.fullname" .) "-alertmanager") .Values.alertmanager.serviceAccount.name }}
 {{- else -}}
     {{ default "default" .Values.alertmanager.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
 
 {{/* Create the name of thanosRuler service account to use */}}
-{{- define "kube-prometheus-stack.thanosRuler.serviceAccountName" -}}
+{{- define "project-prometheus-stack.thanosRuler.serviceAccountName" -}}
 {{- if .Values.thanosRuler.serviceAccount.create -}}
-    {{ default (include "kube-prometheus-stack.thanosRuler.name" .) .Values.thanosRuler.serviceAccount.name }}
+    {{ default (include "project-prometheus-stack.thanosRuler.name" .) .Values.thanosRuler.serviceAccount.name }}
 {{- else -}}
     {{ default "default" .Values.thanosRuler.serviceAccount.name }}
 {{- end -}}
@@ -279,7 +270,7 @@
 {{/*
 Allow the release namespace to be overridden for multi-namespace deployments in combined charts
 */}}
-{{- define "kube-prometheus-stack.namespace" -}}
+{{- define "project-prometheus-stack.namespace" -}}
   {{- if .Values.namespaceOverride -}}
     {{- .Values.namespaceOverride -}}
   {{- else -}}
@@ -290,7 +281,7 @@
 {{/*
 Use the grafana namespace override for multi-namespace deployments in combined charts
 */}}
-{{- define "kube-prometheus-stack-grafana.namespace" -}}
+{{- define "project-prometheus-stack-grafana.namespace" -}}
   {{- if .Values.grafana.namespaceOverride -}}
     {{- .Values.grafana.namespaceOverride -}}
   {{- else -}}
@@ -299,20 +290,9 @@
 {{- end -}}
 
 {{/*
-Allow kube-state-metrics job name to be overridden
-*/}}
-{{- define "kube-prometheus-stack-kube-state-metrics.name" -}}
-  {{- if index .Values "kube-state-metrics" "nameOverride" -}}
-    {{- index .Values "kube-state-metrics" "nameOverride" -}}
-  {{- else -}}
-    {{- print "kube-state-metrics" -}}
-  {{- end -}}
-{{- end -}}
-
-{{/*
 Use the kube-state-metrics namespace override for multi-namespace deployments in combined charts
 */}}
-{{- define "kube-prometheus-stack-kube-state-metrics.namespace" -}}
+{{- define "project-prometheus-stack-kube-state-metrics.namespace" -}}
   {{- if index .Values "kube-state-metrics" "namespaceOverride" -}}
     {{- index .Values "kube-state-metrics" "namespaceOverride" -}}
   {{- else -}}
@@ -323,7 +303,7 @@
 {{/*
 Use the prometheus-node-exporter namespace override for multi-namespace deployments in combined charts
 */}}
-{{- define "kube-prometheus-stack-prometheus-node-exporter.namespace" -}}
+{{- define "project-prometheus-stack-prometheus-node-exporter.namespace" -}}
   {{- if index .Values "prometheus-node-exporter" "namespaceOverride" -}}
     {{- index .Values "prometheus-node-exporter" "namespaceOverride" -}}
   {{- else -}}
@@ -332,13 +312,13 @@
 {{- end -}}
 
 {{/* Allow KubeVersion to be overridden. */}}
-{{- define "kube-prometheus-stack.kubeVersion" -}}
+{{- define "project-prometheus-stack.kubeVersion" -}}
   {{- default .Capabilities.KubeVersion.Version .Values.kubeVersionOverride -}}
 {{- end -}}
 
 {{/* Get Ingress API Version */}}
-{{- define "kube-prometheus-stack.ingress.apiVersion" -}}
-  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19-0" (include "kube-prometheus-stack.kubeVersion" .)) -}}
+{{- define "project-prometheus-stack.ingress.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19-0" (include "project-prometheus-stack.kubeVersion" .)) -}}
       {{- print "networking.k8s.io/v1" -}}
   {{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" -}}
     {{- print "networking.k8s.io/v1beta1" -}}
@@ -348,19 +328,19 @@
 {{- end -}}
 
 {{/* Check Ingress stability */}}
-{{- define "kube-prometheus-stack.ingress.isStable" -}}
-  {{- eq (include "kube-prometheus-stack.ingress.apiVersion" .) "networking.k8s.io/v1" -}}
+{{- define "project-prometheus-stack.ingress.isStable" -}}
+  {{- eq (include "project-prometheus-stack.ingress.apiVersion" .) "networking.k8s.io/v1" -}}
 {{- end -}}
 
 {{/* Check Ingress supports pathType */}}
 {{/* pathType was added to networking.k8s.io/v1beta1 in Kubernetes 1.18 */}}
-{{- define "kube-prometheus-stack.ingress.supportsPathType" -}}
-  {{- or (eq (include "kube-prometheus-stack.ingress.isStable" .) "true") (and (eq (include "kube-prometheus-stack.ingress.apiVersion" .) "networking.k8s.io/v1beta1") (semverCompare ">= 1.18-0" (include "kube-prometheus-stack.kubeVersion" .))) -}}
+{{- define "project-prometheus-stack.ingress.supportsPathType" -}}
+  {{- or (eq (include "project-prometheus-stack.ingress.isStable" .) "true") (and (eq (include "project-prometheus-stack.ingress.apiVersion" .) "networking.k8s.io/v1beta1") (semverCompare ">= 1.18-0" (include "project-prometheus-stack.kubeVersion" .))) -}}
 {{- end -}}
 
 {{/* Get Policy API Version */}}
-{{- define "kube-prometheus-stack.pdb.apiVersion" -}}
-  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">= 1.21-0" (include "kube-prometheus-stack.kubeVersion" .)) -}}
+{{- define "project-prometheus-stack.pdb.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "policy/v1") (semverCompare ">= 1.21-0" (include "project-prometheus-stack.kubeVersion" .)) -}}
       {{- print "policy/v1" -}}
   {{- else -}}
     {{- print "policy/v1beta1" -}}
@@ -368,14 +348,14 @@
   {{- end -}}
 
 {{/* Get value based on current Kubernetes version */}}
-{{- define "kube-prometheus-stack.kubeVersionDefaultValue" -}}
+{{- define "project-prometheus-stack.kubeVersionDefaultValue" -}}
   {{- $values := index . 0 -}}
   {{- $kubeVersion := index . 1 -}}
   {{- $old := index . 2 -}}
   {{- $new := index . 3 -}}
   {{- $default := index . 4 -}}
   {{- if kindIs "invalid" $default -}}
-    {{- if semverCompare $kubeVersion (include "kube-prometheus-stack.kubeVersion" $values) -}}
+    {{- if semverCompare $kubeVersion (include "project-prometheus-stack.kubeVersion" $values) -}}
       {{- print $new -}}
     {{- else -}}
       {{- print $old -}}
@@ -386,21 +366,21 @@
 {{- end -}}
 
 {{/* Get value for kube-controller-manager depending on insecure scraping availability */}}
-{{- define "kube-prometheus-stack.kubeControllerManager.insecureScrape" -}}
+{{- define "project-prometheus-stack.kubeControllerManager.insecureScrape" -}}
   {{- $values := index . 0 -}}
   {{- $insecure := index . 1 -}}
   {{- $secure := index . 2 -}}
   {{- $userValue := index . 3 -}}
-  {{- include "kube-prometheus-stack.kubeVersionDefaultValue" (list $values ">= 1.22-0" $insecure $secure $userValue) -}}
+  {{- include "project-prometheus-stack.kubeVersionDefaultValue" (list $values ">= 1.22-0" $insecure $secure $userValue) -}}
 {{- end -}}
 
 {{/* Get value for kube-scheduler depending on insecure scraping availability */}}
-{{- define "kube-prometheus-stack.kubeScheduler.insecureScrape" -}}
+{{- define "project-prometheus-stack.kubeScheduler.insecureScrape" -}}
   {{- $values := index . 0 -}}
   {{- $insecure := index . 1 -}}
   {{- $secure := index . 2 -}}
   {{- $userValue := index . 3 -}}
-  {{- include "kube-prometheus-stack.kubeVersionDefaultValue" (list $values ">= 1.23-0" $insecure $secure $userValue) -}}
+  {{- include "project-prometheus-stack.kubeVersionDefaultValue" (list $values ">= 1.23-0" $insecure $secure $userValue) -}}
 {{- end -}}
 
 {{/* Sets default scrape limits for servicemonitor */}}
@@ -437,7 +417,7 @@
   - pullSecret1
   - pullSecret2
 */}}
-{{- define "kube-prometheus-stack.imagePullSecrets" -}}
+{{- define "project-prometheus-stack.imagePullSecrets" -}}
 {{- range .Values.global.imagePullSecrets }}
   {{- if eq (typeOf .) "map[string]interface {}" }}
 - {{ toYaml . | trim }}
@@ -447,9 +427,28 @@
 {{- end }}
 {{- end -}}
 
-{{- define "kube-prometheus-stack.operator.admission-webhook.dnsNames" }}
-{{- $fullname := include "kube-prometheus-stack.operator.fullname" . }}
-{{- $namespace := include "kube-prometheus-stack.namespace" . }}
+
+{{/* Define ingress for all hardened namespaces */}}
+{{- define "project-prometheus-stack.hardened.networkPolicy.ingress" -}}
+{{- $root := index . 0 }}
+{{- $ns := index . 1 }}
+{{- if $root.Values.global.networkPolicy.ingress -}}
+{{ toYaml $root.Values.global.networkPolicy.ingress }}
+{{- end }}
+{{- if $root.Values.global.networkPolicy.limitIngressToProject }}
+- from:
+{{- if $root.Values.global.cattle.projectNamespaceSelector }}
+  - namespaceSelector: {{- $root.Values.global.cattle.projectNamespaceSelector | toYaml | nindent 6 }}
+{{- end }}
+  - namespaceSelector:
+      matchLabels:
+        kubernetes.io/metadata.name: {{ $ns }}
+{{- end }}
+{{- end -}}
+
+{{- define "project-prometheus-stack.operator.admission-webhook.dnsNames" }}
+{{- $fullname := include "project-prometheus-stack.operator.fullname" . }}
+{{- $namespace := include "project-prometheus-stack.namespace" . }}
 {{- $fullname }}
 {{ $fullname }}.{{ $namespace }}.svc
 {{- if .Values.prometheusOperator.admissionWebhooks.deployment.enabled }}
